### PR TITLE
Project deploy changes

### PR DESCRIPTION
**Pull request by anthmatic**
## Description and Context

- [x]  Update `hs project deploy` for new deploy service
  - [x] add new endpoints
  - [x] poll for deployed status

### CODE REVIEW

File: packages/cli/lib/projects.js [hunk 1]
```diff
@@ -232,11 +232,102 @@ const pollBuildStatus = async (accountId, name, buildId) => {
   });
 };
 
+const pollDeployStatus = async (accountId, name, deployId) => {
```
Comment (anthmatic) [inline packages/cli/lib/projects.js:235]: This is very similar to the build status polling logic. Will definitely DRY this up- time permitting, I'll do that in this PR

**Comment by brandenrodgers** [comment]
I haven't dug into this yet, but I wanted to mention that IMO it's OK to get this deploy work out without handling the " Update hs project upload to be aware of deployments" work. If you want you could handle that in a separate PR.

**Review by miketalley — APPROVED** [review]
Code looks good!

**Comment by anthmatic** [comment]
Example of the manual deploy flow:
image

Since the auto-deploy side of things is still in flight I'm going to get this in so that folks can test.

**Pull request merged**